knotx-core/knotx-fragment-assembler/src/main/java/com/cognifide/knotx/knot/assembler/FragmentAssemblerVerticle.java

```diff
@@ -0,0 +1,124 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.knotx.knot.assembler;
+
+import com.cognifide.knotx.dataobjects.ClientResponse;
+import com.cognifide.knotx.dataobjects.KnotContext;
+import com.cognifide.knotx.fragments.Fragment;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.rxjava.core.AbstractVerticle;
+import io.vertx.rxjava.core.MultiMap;
+import io.vertx.rxjava.core.buffer.Buffer;
+import io.vertx.rxjava.core.eventbus.EventBus;
+import io.vertx.rxjava.core.eventbus.Message;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import rx.Observable;
+
+public class FragmentAssemblerVerticle extends AbstractVerticle {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FragmentAssemblerVerticle.class);
+
+  private static final String SNIPPET_IDENTIFIER_NAME = "data-knot-types";
+
+  private static final String ANY_SNIPPET_PATTERN =
+      "(?is).*<script\\s+" + SNIPPET_IDENTIFIER_NAME + ".*";
+
+  private FragmentAssemblerConfiguration configuration;
+
+  @Override
+  public void init(Vertx vertx, Context context) {
+    super.init(vertx, context);
+    this.configuration = new FragmentAssemblerConfiguration(config());
+  }
+
+  @Override
+  public void start() throws IOException, URISyntaxException {
+    LOGGER.debug("Starting <{}>", this.getClass().getName());
+    EventBus eventBus = vertx.eventBus();
+
+    eventBus.<KnotContext>consumer(configuration.getAddress()).handler(
+        message -> Observable.just(message)
+            .doOnNext(this::traceMessage)
+            .subscribe(
+                response -> {
+                  KnotContext context = response.body();
+                  response.reply(createSuccessResponse(context, getTemplateContent(context)));
+                },
+                error -> {
+                  LOGGER.error("Exception happened during HTML splitting.", error);
+                  message.reply(processError(message.body(), error));
+                }
+            )
+    );
+  }
+
+  private String getTemplateContent(KnotContext context) {
+    return context.fragments().map(fragments -> fragments.stream()
+        .map(Fragment::getContent)
+        .filter(StringUtils::isNotBlank)
+        .map(value -> {
+          if (value.matches(ANY_SNIPPET_PATTERN)) {
```

Comment:
Do we need this check here ? I thought that Knots for each type suppose to do it ?